src/main/java/FuncTest/TestFunc/Function.java

```diff
@@ -32,21 +35,64 @@ public HttpResponseMessage run(
 		c.getLogger().info("=========== WEBHOOK INVOKED ===========");
 		JSONObject queryResult = new JSONObject(s.getBody().get().toString()).getJSONObject("queryResult");
 		switch (queryResult.getJSONObject("intent").getString("displayName")) {
-      case globals.BUSINESS_HOUR_BY_DAY_INTENT_NAME:
-        return getHourByDay(queryResult, s, c);
-      case globals.BUSINESS_HOUR_WEEK_INTENT_NAME:
-        return getHourByWeek(queryResult, s, c);
-      case globals.FILTER_COURSES_INTENT_NAME:
-        return getMatchingCoursesResponse(queryResult, s, c);
-      case globals.HOMEWORK_GET_UPCOMING_INTENT_NAME:
-        return getUpcomingHomework(queryResult, s, c);
+			case globals.BUSINESS_HOUR_BY_DAY_INTENT_NAME:
+				return getHourByDay(queryResult, s, c);
+			case globals.BUSINESS_HOUR_WEEK_INTENT_NAME:
+				return getHourByWeek(queryResult, s, c);
+			case globals.HOMEWORK_GET_UPCOMING_INTENT_NAME:
+				return getUpcomingHomework(queryResult, s, c);
+			case globals.LOGIN_INTENT:
+				return checkLoginName(queryResult, s, c);
+			case globals.SUBSCRIBE_INTENT:
+				return subscribeToSystem(queryResult,s,c);
+			case globals.FILTER_COURSES_INTENT_NAME:
+				return getMatchingCoursesResponse(queryResult, s, c);
+			case globals.RUN_RULES_INTENT:
+				return applyRules(queryResult, s, c);
       case globals.COURSE_GET_POSTREQUISITES_BY_NAME_INTENT_NAME:
          return PostrequisiteHandler.getPostrequisitesByName(queryResult, s, c);
       case globals.COURSE_GET_POSTREQUISITES_BY_NUMBER_INTENT_NAME:
         return PostrequisiteHandler.getPostrequisitesByNumber(queryResult, s, c);
 		}
 		return utils.createWebhookResponseContent("what is this intent?", s);
 	}
+	
+	private HttpResponseMessage applyRules(JSONObject queryResult, HttpRequestMessage<Optional<String>> s,
+			ExecutionContext c) {
+		// TODO Auto-generated method stub
+		String uname = utils.getStringUserParamFromContext(queryResult, "username");
+		String passwd = utils.getStringUserParamFromContext(queryResult, "password");
+		
+		c.getLogger().info("================ username = " + uname + "================");
```

Comment:
Please remove before final merge of the branch

Reply:
I've removed the logging

Reply:
Done